src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationContext.cs

```diff
@@ -100,21 +98,43 @@ public AuthenticationContext(string authority, TokenCache tokenCache)
         /// </summary>
         /// <param name="authority">Address of the authority to issue token.</param>
         /// <param name="validateAuthority">Flag to turn address validation ON or OFF.</param>
-        /// <param name="tokenCache">Token cache used to lookup cached tokens on calls to AcquireToken</param>
+        /// <param name="tokenCache">Token cache used to lookup cached tokens on calls to AcquireToken. Use <see cref="TokenCache.DefaultShared"/> 
+        /// to use ADAL's implementation of the cache on Android, iOS and UWP. Use null to store tokens only in memory. </param>
         public AuthenticationContext(string authority, bool validateAuthority, TokenCache tokenCache)
             : this(null, authority, validateAuthority ? AuthorityValidationType.True : AuthorityValidationType.False,
                 tokenCache)
         {
         }
 
+        /// <summary>
+        /// Constructor to create the context with the address of the authority, flag to turn address validation off
+        /// and custom HttpClient.
+        /// Make sure you are aware of the security implication of not validating the address.
+        /// </summary>
+        /// <remarks>See https://aka.ms/adal-custom-httpclient for details</remarks>
+        /// <param name="authority">Address of the authority to issue token.</param>
+        /// <param name="validateAuthority">Flag to turn address validation ON or OFF.</param>
+        /// <param name="tokenCache">Token cache used to lookup cached tokens on calls to AcquireToken. Use <see cref="TokenCache.DefaultShared"/> 
+        /// to use ADAL's implementation of the cache on Android, iOS and UWP. Use null to store tokens only in memory.</param>
+        /// <param name="httpClient">Custom Http Client that will be used for the lifetime of this object</param>
+        public AuthenticationContext(string authority, bool validateAuthority, TokenCache tokenCache, HttpClient httpClient)
+            : this(null, authority, validateAuthority ? AuthorityValidationType.True : AuthorityValidationType.False,
+                   tokenCache, httpClient)
+        {
+            if (httpClient == null)
```

Comment:
If someone is using this ctor, they better pass in an httpClient. keeps things tidy.